### DISCUSSION TITLE
Why sending comments raise an error except when sending to myself?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#13916] [Archive]
1. While typing the comment on a detail, when I type `@` I see all the users.
But when I choose another user than myself, sending the comment stops with 
```
{
  "errors": [
    {
      "message": "An unexpected error occurred.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```
_The user has all rights on Add/Read/Write for Directus_notifications and sees the complete users list._
What am I missing for comments?

-

2. With share, when I click on `Shares`, I don't have the button `New Share` as shown in the docs
_The user has all rights on Add/Read/Write for Directus_shares and sees the complete users list._
What am I missing for shares?

**Comment by rijkvanzanten** [comment]
Can you try this as an admin? That should return more elaborate information in the returned error. Alternatively, check if your server logs indicate of anything 🤔

**Reply by Pachat** [reply]
As admin, it gives the message
```
{
  "errors": [
    {
      "message": "spawn /usr/sbin/sendmail ENOENT",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```
and after an unsuccessful try for any subsequent try, even when elevating the receiver to Admin and even when an admin tries to send to oneself.


As for `Share` to have the button `New Share` was only a question of activating `Share` in the role. 
So this part of the question is solved.

**Reply by azrikahar** [reply]
Since it will only seemingly work for you and based on the error shown, I presume you have `Email Notifications` disabled in your own profile's User Preference? Ref https://github.com/directus/directus/discussions/12997#discussioncomment-2630977. Other users have it enabled (by default) which may be why comments subsequently triggers email notifications and throw that error.

**Reply by Pachat** [reply]
Thanks for pointing that. But all the users, including me have `Email Notifications` enabled.
Must be something else.

Could it be that I am on localhost, developing locally on my computer?
Is there anything to set concerning sendmail?
BTW why internal to the App notifications try to sendmail?

**Reply by rijkvanzanten** [reply]
> Could it be that I am on localhost, developing locally on my computer?

Yup! Looks like you don't have `sendmail` available in your path

> Is there anything to set concerning sendmail?

Yeah, you can install sendmail locally, or use a service like mailtrap if you need a debug SMTP server

> BTW why internal to the App notifications try to sendmail?

Not quite sure what you're asking.. If you're asking as to why the app would try to send an email when you're creating a notification, it's because of the aforementioned email notifications. If you tag a user, Directus will send them an email notifying of the mention 👍🏻
